web-console/src/dialogs/compaction-dialog/compaction-dialog.spec.tsx

```diff
@@ -21,15 +21,28 @@ import React from 'react';
 
 import { CompactionDialog } from './compaction-dialog';
 
-describe('compaction dialog', () => {
-  it('matches snapshot', () => {
+describe('CompactionDialog', () => {
+  it('matches snapshot without compactionConfig', () => {
     const compactionDialog = shallow(
       <CompactionDialog
         onClose={() => {}}
         onSave={() => {}}
         onDelete={() => {}}
-        datasource={'test'}
-        compactionConfig={{}}
+        datasource={'test1'}
+        compactionConfig={undefined}
+      />,
+    );
+    expect(compactionDialog).toMatchSnapshot();
+  });
+
+  it('matches snapshot with compactionConfig', () => {
```

Comment:
What do you think about having 3 variants of this test: `dynamic`, `hashed`, and `single_dim` partitions. For example, the snapshot would verify that the "Max num merge tasks" field should only show up for `hashed` and `single_dim`. Also, for `single_dim` we could have a test to verify that the spec is valid (see the comment I left on `validCompactionConfig()`).

Reply:
👍